### DISCUSSION TITLE
Why do the samples contain 'Math.random()'?

### DISCUSSION THREAD

**Discussion by onionhammer** [airbnb/visx#1754] [Q&A] [answered]
Just saw this pop up in sonarcloud for me, and was wondering what the reasoning was for random key/ids to be added in the samples

![image](https://github.com/airbnb/visx/assets/969938/be269812-3ad2-462b-a01a-0d66482085f9)

Is this to force re-rendering? Does it break without random keys?

**Comment by williaster** [comment, marked as answer]
From the docs https://airbnb.io/visx/docs/threshold#Threshold_id

> id for this threshold. If not set, multiple Threshold's on a page may conflict and interfere with each other.

thresholds use `clip-path`s which require `id`s. in examples, it doesn't matter if this is random. in your application you could consider using a unique id that is stable across renders. a simple solution (that would arguably demonstrate better react practices) would be to memoize this value with `useMemo/useRef`.

**Reply by onionhammer** [reply]
or `useId()` perhaps
